main.py

```diff
@@ -55,95 +54,81 @@ def main():
-            elif to_trim:
-                # Get the start and end trim in points from the user.
-                in_point = set_timestamp(
-                    "Please specify the trim 'in' point")
-                out_point = set_timestamp(
-                    "Please specify the trim 'out' point")
-
-                if to_watermark:
-                    # Make the ffmpeg subprocess call.
-                    ffmpeg_call = [
-                        'ffmpeg', '-ss', in_point,
-                        '-to', out_point,
-                        '-i', sys.argv[1],
-                        '-i', watermark_path(aspect_ratio),
-                        '-filter_complex', "[0]yadif=0:-1:1[v];[v][1]overlay",
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750', 
-                        '-c:a', 'aac',                       
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                else:
-                    # Make the ffmpeg subprocess call.
-                    ffmpeg_call = [
-                        'ffmpeg', '-ss', in_point,
-                        '-to', out_point,
-                        '-i', input_file,
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-vf', 'yadif',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750',                        
-                        '-c:a', 'aac',                        
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                    
+            ffmpeg_command = create_ffmpeg_command(input_file)
+
+            print("Transcoding with:", " ".join(ffmpeg_command), "\n")
+
+            subprocess.call(ffmpeg_command)
+
+
+def create_ffmpeg_command(input_file):
+    output_path = set_output_path(input_file)
+
+    to_watermark = set_watermark_status()
+    to_trim = set_trim_status()
+
+    aspect_ratio = stream_aspect_ratio(input_file)
+    dar = display_aspect_ratio(aspect_ratio)
+    watermark_file = watermark_path(aspect_ratio)
+
+    ffmpeg_program_call = [
+        "ffmpeg"
+    ]
+
+    input_video_file = [
+        "-i", input_file
+    ]
+
+    watermark_settings = [
+        "-i", watermark_file,
+        "-filter_complex", "[0]yadif=0:-1:1[v];[v][1]overlay"  # overlay filter for supplied png
+    ]
+
+    stream_settings = [
+        # TODO "setdar=dar={0}".format(dar),   # set DAR to aspect ratio
```

Comment:
Not sure what's going on with `setdar` - when I had this in the script, I was getting the error
```bash
Unable to find a suitable output format for 'setdar=dar=1.3333333333333333'
setdar=dar=1.3333333333333333: Invalid argument
```
The ffmpeg [docs](https://ffmpeg.org/ffmpeg-filters.html#Examples-97) give an example of `setdar=dar=1.77777`, so it could be too many decimal places? Or a string v. int thing. Anyway, I leave this to later to debug :)

Reply:
I realised what the deal is here - `setdar` is not a standalone thing, it's part of the video filter. Will fix in a follow-up commit

Reply:
If it's of any use (and I've not really looked at the rest of this thread) you can use ratios e.g. setdar=4/3

Reply:
Set the DAR in b508b7b. If there's an overlay (watermark), that needs to be part of the same filtergraph, so that commit includes logic to combine them if necessary

Reply:
I just saw your comment @Warblefly! I hadn't refreshed the page. Yep, both floats (`1.333(3)`) and ratios (`4:3`) work here. What I was doing wrong was that I forgot that `setdar` was a filter, not a top-level command, opps.  
Changed in b508b7b, would love feedback if you have it!